crates/solvers/src/domain/solver/baseline.rs

```diff
@@ -27,6 +28,10 @@ pub struct Baseline {
     /// - A value of 2 indicates: `A -> B -> C -> D`
     /// - etc.
     pub max_hops: usize,
+    /// The maximum number of attempts to solve a partially fillable order.
+    /// Basically we continuously halve the amount to execute until we find a
```

Comment:
We can do the proper binary search for the baseline solver over the simpler approach that was needed when solving across multiple `/solve` requests.
The reason is that the binary search requires the search space to be sorted in order to produce a correct result (or generally make correct progress). This is not the case when we search for the ideal `fill_amount` across multiple `/solve` requests as the ideal `fill_amount` can move as liquidity moves around when more blocks get built. This is not the case for the baseline solver as it always operates on a single block which makes the search space stable and therefore sorted so to speak.
IMO this shouldn't block merging this PR as we are short on time but it's an improvement we should consider in the future.

Reply:
> IMO this shouldn't block merging this PR as we are short on time but it's an improvement we should consider in the future.

I though of this, but unfortunately we are trying to optimize for surplus which isn't increasing and decreasing over the search space.

I agree there are other improvements that we can do, but given how the baseline solver also has quite limited liquidity I'm not sure it is worth the effort (without something like an LP book to back it up).

Reply:
> I though of this, but unfortunately we are trying to optimize for surplus which isn't sorted over the search space.

That's true. Didn't think about maximizing surplus. 🤦‍♂️ 
> but given how the baseline solver also has quite limited liquidity I'm not sure it is worth the effort.

💯